edx_proctoring/views.py

```diff
@@ -286,6 +286,72 @@ def get(self, request, exam_id=None, course_id=None, content_id=None):  # pylint
         return Response(data)
 
 
+class StudentOnboardingStatusView(ProctoredAPIView):
+    """
+    Endpoint for the StudentOnboardingStatusView
+
+    Supports:
+        HTTP GET: returns the learner's onboarding status relative to the given course_id
```

Comment:
I can see we need a new endpoint to retrieve the onboarding status for a single user regardless of their course. I also believe the endpoint needs permission logic to ensure, either the request user is a course admin user, edX staff user, or the passed in user is the same user on the request. All other request should be denied.
Please make sure you setup your unit tests for these 4 permission cases.

Reply:
I added logic to pass in a username, and to deny access if the username doesn't match the logged in user + the user is also not course/edX staff. If a username isn't given, the endpoint just returns the logged in user's own status.